packages/abstraction/test/SchemaValidator.test.ts

```diff
@@ -0,0 +1,73 @@
+import path from 'path';
+import { validateSchema } from "../src/SchemaValidator";
+import { ContractGenerator } from "../src";
+import { MultipleContractAdapter, EOSContractAdapter, ICONContractAdapter, ModelContractAdapter } from "@aloxide/bridge";
+
+describe('test SchemaValidator', () => {
+
+  let contractGenerator;
+  const logger = {
+    info: jest.fn(),
+    debug: jest.fn(),
+    error: jest.fn(),
+    success: jest.fn(),
+  };
+  const adapter = new MultipleContractAdapter();
+
+  adapter.addAdapters(
+    new EOSContractAdapter(),
+    new ICONContractAdapter(),
+    new ModelContractAdapter(),
+  );
+
+  const checkName = val => {
+    return /^[1-5a-zA-Z]+/.test(val)
+  }
+
+  const checkType = val => {
+    const supportedType = ["uint64_t", "number", "string", "array", "bool"];
+    return supportedType.indexOf(val) !== -1
+  }
+
+  const requiredSchema = {
+    entities: [{
+      name: { type: String, required: true, length: { min: 1, max: 12 }, use: { checkName } },
+      fields: [{
+        name: { type: String, required: true, length: { min: 1, max: 12 }, use: { checkName } },
+        type: { type: String, required: true, use: { checkType } },
+      }],
+      key: { type: String, required: true, length: { min: 1, max: 12 }, use: { checkName } }
+    }
+    ]
+  };
```

Comment:
I see above lines repeated in another test, consider to reuse them.

Reply:
We need to refactor some functions because define function in function so hard to test

Reply:
Can you move above code to a function and just call it here? So the function can be reused in other tests?

Reply:
updated